profiler/data.py

```diff
@@ -1,51 +1,63 @@
 import errno
+import glob
+import hashlib
+import json
 import os
-import pickle
-import re
 from abc import ABC, abstractmethod
-from typing import Dict, List, Optional
+from typing import Any, Dict, List, Optional
 
-from attr import attrib, attrs
+import attr
 
-"""This class represents the data stored per run"""
 
-
-@attrs
+@attr.define
 class ProfileData:
-    process: str = attrib()
-    custom_out: List[Optional[str]] = attrib()
-    rt: float = attrib()
-    ut: float = attrib()
-    st: float = attrib()
-    max_set_size: int = attrib()
-    page_reclaims: int = attrib()
-    page_faults: int = attrib()
-    cycles_elapsed: int = attrib()
-    peak_memory: int = attrib()
-    tiledb_stats: Optional[str] = attrib()
-    date: str = attrib()
-    now: str = attrib()
-    somacore_version: str = attrib()
-    tiledbsoma_version: str = attrib()
-    context: Dict[str, str] = attrib()
-    print("Creating ProfileData")
-
-
-def extract_key_from_filename(filename: str) -> str:
-    """Extracts DB key for stats from the corresponding filename"""
-    match = re.match("(.+)\.run", filename)
-    assert match
-    return match.groups()[0]
-
-
-PROFILE_PATH = "./profiling_runs"
-
-
-def improve_profileDB_key(process: str) -> str:
-    """Remove space characters from profileDB keys."""
-    name: str = process.replace(" ", "_").replace("/", "_").replace(".", "_")
-    print(f"Profiler key = {name}")
-    return name
+    """This class represents the data stored per run"""
+
+    command: str
+    timestamp: float
+    stdout: str
+    stderr: str
+    tiledb_stats: Dict[str, Any]
+    somacore_version: str
+    tiledbsoma_version: str
+    host_context: Dict[str, str]
+    user_time_sec: float
+    system_time_sec: float
+    pct_of_cpu: float
+    elapsed_time_sec: float
+    avg_shared_text_sz_kb: int
+    avg_unshared_text_sz_kb: int
+    avg_stack_sz_kb: int
+    avg_total_sz_kb: int
+    max_res_set_sz_kb: int
+    avg_res_set_sz_kb: int
+    major_page_faults: int
+    minor_page_faults: int
+    voluntary_context_switches: int
+    involuntary_context_switches: int
+    swaps: int
+    file_system_inputs: int
+    file_system_outputs: int
+    socket_messages_sent: int
+    socket_messages_received: int
+    signals_delivered: int
+    page_size_bytes: int
+    exit_status: int
+    custom_out: List[Optional[str]]
+
+    command_key: str = attr.field()
```

Comment:
What does command_kay do and what exactly does this decorator do?

Reply:
this is a hash of the command, which allows storing the profiling data under a name that has a bounded length; the commands I was using to profile ended up being longer than allowed by the filesystem.